azure-cli/monitor/alert.utilities.ps1

```diff
@@ -0,0 +1,259 @@
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $resourceGroupName,
+    
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string] 
+        $serviceName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string] 
+        $resourceType
+    )
+    $resourceId = az resource show `
+        --resource-group $resourceGroupName `
+        --name $serviceName `
+        --resource-type $resourceType `
+        --query id
+    return $resourceId.Replace('"','')
+}
+
+function New-ActionGroup{
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $resourceGroupName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string] 
+        $actionGroupName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string] 
+        $actionGroupShortName,
+
+        [Parameter(Mandatory=$false)]
+        [string[]] 
+        $tags = @()
+    )
+
+    return az monitor action-group create `
+        --resource-group $resourceGroupName `
+        --name $actionGroupName `
+        --short-name $actionGroupShortName `
+        --tags $tags
+}
+
+function Add-EmailAction{
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $resourceGroupName,
+    
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string] 
+        $actionGroupName,
+        
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string] 
+        $emailName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string] 
+        $emailAddress
+    )
+
+    return az monitor action-group update `
+    --resource-group $resourceGroupName `
+    --name $actionGroupName `
+    --add-action email $emailName $emailAddress 
+}
+
+function Get-EmailAction{
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $resourceGroupName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $actionGroupName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $emailAddress
+    )
+
+    return az monitor action-group show `
+        --resource-group $resourceGroupName `
+        --name $actionGroupName `
+        --query "emailReceivers[?(emailAddress == '$emailAddress')]" `
+        | ConvertFrom-Json
+}
+
+function Get-AllEmailAction{
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $resourceGroupName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $actionGroupName
+    )
+
+    return az monitor action-group show `
```

Comment:
could we use list with some JMESPath query instead? show is rather slow..

Reply:
Changed.